roles/openshift_node_dnsmasq/files/networkmanager/99-origin-dns.sh

```diff
@@ -61,35 +59,40 @@ EOF
       NEEDS_RESTART=1
     fi
 
-    ######################################################################
-    # Write out default nameservers for /etc/dnsmasq.d/origin-upstream-dns.conf
-    # and /etc/origin/node/resolv.conf in their respective formats
-    for ns in ${IP4_NAMESERVERS}; do
-      if [[ ! -z $ns ]]; then
-        echo "server=${ns}" >> $UPSTREAM_DNS_TMP
-        echo "nameserver ${ns}" >> $NEW_NODE_RESOLV_CONF
+    # If network manager doesn't know about the nameservers then the best
+    # we can do is grab them from /etc/resolv.conf but only if we've got no
+    # watermark
+    if ! grep -q '99-origin-dns.sh' /etc/resolv.conf; then
+      if ! [[ -n "${IP4_NAMESERVERS}" && "${IP4_NAMESERVERS}" != "${def_route_ip}" ]]; then
```

Comment:
can't this be rewritten as `if [[ -z "${IP4_NAMESERVERS}" || "${IP4_NAMESERVERS}" == "${def_route_ip}" ]]; then` ? that seems more obvious to me as to what we're checking for rather than negating the `if` afterwards...